openrelik_worker_common/mount_utils.py

```diff
@@ -0,0 +1,266 @@
+        mountpoints = bd.mount()
+        # Do the things you need to do :)
+        bd.destroy()
+    """
+
+    def __init__(self, image_path: str):
+        """Initialize BlockDevice class instance.
+
+        Args:
+            image_path (str): path to the image file to map and mount.
+        """
+        self.image_path = image_path
+        self.blkdevice = None
+        self.blkdeviceinfo = None
+        self.partitions = []
+        self.mountpoints = []
+        self.mountroot = "/mnt"
+
+        # Setup the loop device
+        self._losetup()
+
+        # Parse block device info
+        self._blkinfo()
+
+        # Parse partition information
+        self._parse_partitions()
+
+    def _losetup(self):
+        """Map image file to loopback device using losetup.
+
+        Returns: None
+        """
+        losetup_command = [
+            "sudo",
+            "losetup",
+            "--find",
+            "--partscan",
+            "--show",
+            self.image_path,
+        ]
+
+        process = subprocess.run(
+            losetup_command, capture_output=True, check=False, text=True
+        )
+        if process.returncode == 0:
+            self.blkdevice = process.stdout.strip()
+        else:
+            raise RuntimeError(f"Error: {process.stderr} {process.stdout}")
+
+        return None
+
+    def _blkinfo(self):
+        """Extract device and partition information using blkinfo.
+
+        Returns: None
+        """
+        lsblk_command = ["sudo", "lsblk", "-ba", "-J", self.blkdevice]
+
+        process = subprocess.run(
+            lsblk_command, capture_output=True, check=False, text=True
+        )
+        if process.returncode == 0:
+            self.blkdeviceinfo = json.loads(process.stdout.strip())
+        else:
+            raise RuntimeError(f"Error lsblk:  {process.stderr} {process.stdout}")
+
+        return None
+
+    def _parse_partitions(self):
+        """Parse partition information from block device details.
+
+        Returns: None
+        """
+        bd = self.blkdeviceinfo.get("blockdevices")[0]
+        if "children" not in bd:
+            # No partitions on this disk.
+            return
+        for children in bd.get("children"):
+            partition = f"/dev/{children["name"]}"
+            if self._is_important_partition(children):
+                self.partitions.append(partition)
+
+        return None
+
+    def _is_important_partition(self, partition: str):
+        """Decides if we will process a partition. We process the partition if:
+        * > 100Mbyte in size
+        * contains a filesystem type ext*, dos, vfat, xfs, ntfs
+
+        Args:
+            partitions (str): Name of the partition.
+
+        Returns:
+            bool: True or False for importance of partition.
+        """
+        if partition["size"] < 100000000:
+            return False
+        fs_type = self._get_fstype(f"/dev/{partition["name"]}")
+        if fs_type not in [
```

Comment:
I would make this a class attribute or Enum and check if fs_type in self.supported_fstypes.

Reply:
Should we make `100000000` some sort of class attribute that could be changed more easily if this ends up being used in other places?